backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/impl/ElasticsearchSearchQueryImpl.java

```diff
@@ -149,7 +146,21 @@ public long fetchTotalHitCount() {
 
 	@Override
 	public SearchScroll<H> scroll(Integer pageSize) {
-		throw new UnsupportedOperationException( "Not yet implemented" );
+		NonBulkableWork<ElasticsearchLoadableSearchResult<H>> work = getSearchWorkBuilder()
+				.scrolling( pageSize, "60s" )
+				.build();
+
+		final ElasticsearchSearchResult<H> searchResult = Futures.unwrappedExceptionJoin( queryOrchestrator.submit( work ) )
+				.loadBlocking();
```

Comment:
I think this should be done on the first call to `next()`, not here. Otherwise you're pre-fetching entities, which may be unexpected to some users: what if they do a `session.clear()` before the call to `scroll()` and the call to `next()`?

I understand this will complicate the implementation of `ElasticsearchSearchScroll`, but I really think it's the only way to comply with the principle of least surprise.

Reply:
Agreed... even if it will complicate a bit the alignment between the scroll instances client side (ours) and the server side (ES), it will be better to involve the server side scroll at the first `next()` time